File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/MainActivity.java
package com.avidprogrammers.insurancepremiumcalculator;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.os.Handler;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Toast;

/**
 * Created by Abhishek on 25-Aug-17.
 */

public class MainActivity extends Activity implements ConnectivityReceiver.ConnectivityReceiverListener {

    ConnectivityReceiver conn;

    CheckingStatus checkingStatus;
    static int status=1;

    // Splash screen timer
    private static int SPLASH_TIME_OUT = 4000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(getIntent().getBooleanExtra("EXIT",false)){
            finishAndRemoveTask();
        }
        checkingStatus=new CheckingStatus();
        conn=new ConnectivityReceiver();
        IntentFilter intentFilter=new IntentFilter();
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(conn, intentFilter);
        checkfunction(MainActivity.this);

        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_main);

        splash();
    }

    public void checkfunction(Context context){
        boolean isConnected=ConnectivityReceiver.isConnected();
        checkingStatus.notification(isConnected, context);
        if(isConnected){
            status=1;
        }else{
            status=0;
        }
    }
    public void splash(){
        new Handler().postDelayed(new Runnable() {

            /*
             * Showing splash screen with a timer. This will be useful when you
             * want to show case your app logo / company
             */
            @Override
            public void run() {
                // This method will be executed once the timer is over
                // Start your app main activity
                if(status==1) {
                    Intent i = new Intent(MainActivity.this, home_activity.class);
                    startActivity(i);
                    finish();
                }else{
                    Toast.makeText(MainActivity.this, "Check Internet Connection !", Toast.LENGTH_SHORT).show();
                }
            }
        }, SPLASH_TIME_OUT);
    }

    @Override
    protected void onResume() {
        super.onResume();
        MyApplication.getInstance().setConnectivityListener(this);
    }

    @Override
    public void onNetworkConnectionChanged(boolean isConnected) {
        checkingStatus.notification(isConnected,this);
        if(isConnected){
            status=1;
        }else{
            status=0;
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(conn);
    }
}

File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/MainActivity.kt
package com.avidprogrammers.insurancepremiumcalculator

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import android.os.Handler
import android.view.Window
import android.view.WindowManager
import android.widget.Toast
import com.avidprogrammers.ads.AppOpenManager
import com.avidprogrammers.insurancepremiumcalculator.ConnectivityReceiver.ConnectivityReceiverListener


/**
 * Created by Abhishek on 25-Aug-17.
 */
class MainActivity : Activity(), ConnectivityReceiverListener {
    var conn: ConnectivityReceiver? = null
    var checkingStatus: CheckingStatus? = null
    var appOpenManager: AppOpenManager? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (intent.getBooleanExtra("EXIT", false)) {
            finishAndRemoveTask()
        }
        checkingStatus = CheckingStatus()
        conn = ConnectivityReceiver()
        val intentFilter = IntentFilter()
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(conn, intentFilter)
        checkfunction(this@MainActivity)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
        setContentView(R.layout.activity_main)

        appOpenManager = AppOpenManager(applicationContext as MyApplication)

        splash()
    }

    fun checkfunction(context: Context?) {
        val isConnected: Boolean = ConnectivityReceiver.Companion.isConnected
        checkingStatus!!.notification(isConnected, context!!)
        if (isConnected) {
            MainActivity.Companion.status = 1
        } else {
            MainActivity.Companion.status = 0
        }
    }

    fun splash() {
        Handler().postDelayed({ // This method will be executed once the timer is over
            // Start your app main activity
            if (MainActivity.Companion.status == 1) {
                val i = Intent(this@MainActivity, home_activity::class.java)
                startActivity(i)
                finish()
                appOpenManager!!.showAdIfAvailable()
            } else {
                Toast.makeText(this@MainActivity, "Check Internet Connection !", Toast.LENGTH_SHORT)
                    .show()
            }
        }, MainActivity.Companion.SPLASH_TIME_OUT.toLong())
    }

    override fun onResume() {
        super.onResume()
        MyApplication.Companion.instance!!.setConnectivityListener(this)
    }

    override fun onNetworkConnectionChanged(isConnected: Boolean) {
        checkingStatus!!.notification(isConnected, this)
        if (isConnected) {
            MainActivity.Companion.status = 1
        } else {
            MainActivity.Companion.status = 0
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(conn)
    }

    companion object {
        var status = 1

        // Splash screen timer
        private const val SPLASH_TIME_OUT = 4000
    }
}